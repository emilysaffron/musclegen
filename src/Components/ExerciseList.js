import styled from "@emotion/styled";
import GetExerciseNames from "../Helpers/GetExerciseNames";
const StyledList = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ExerciseList = ({ exercises, toggleModal,modal }) => {
  const names = GetExerciseNames(exercises, toggleModal,modal);

  return <StyledList>{names}</StyledList>;
};
export default ExerciseList;
