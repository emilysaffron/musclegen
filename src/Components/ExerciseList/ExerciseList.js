/* eslint-disable react-hooks/exhaustive-deps */
import styled from "@emotion/styled";
import GetExerciseNames from "../../Helpers/GetExerciseNames";
import { useState, useEffect } from "react";
import FilterModal from "../FilterModal/FilterModal";

const StyledList = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
const StyledContent = styled.div`
  display: flex;
`;
const ExerciseList = ({
  exercises,
  toggleModal,
  modal,
  updateChosenExercise,
  half,
}) => {
  const [filter, applyFilter] = useState("");
  const [names, updateNames] = useState("");
  const filterExercises = (label) => {
    applyFilter(label);
  };
  useEffect(() => {
    updateNames(
      GetExerciseNames(
        exercises,
        toggleModal,
        modal,
        updateChosenExercise,
        filter
      )
    );
  }, [filter, exercises]);

  return (
    <StyledContent>
      {FilterModal(filterExercises, half)}
      <StyledList>{names}</StyledList>
    </StyledContent>
  );
};
export default ExerciseList;
