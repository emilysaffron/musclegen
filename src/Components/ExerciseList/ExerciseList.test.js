import renderer from "react-test-renderer";
import ExerciseList from "./ExerciseList";

it("ExerciseList Snapshot", () => {
  const tree = renderer
    .create(
      <ExerciseList
        exercises=""
        toggleModal="toggleModal"
        modal={true}
        updateChosenExercise="updateChosenExercise"
        half=""
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
