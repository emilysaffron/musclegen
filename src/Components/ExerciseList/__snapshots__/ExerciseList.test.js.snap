// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseList Snapshot 1`] = `
<div
  className="css-ov1ktg"
>
  <button
    className="css-15i479a"
    onClick={[Function]}
  >
    Filter Exercises
  </button>
  <ul
    className="css-1oskuwq"
  >
    
  </ul>
</div>
`;
